RecoLocalCalo/EcalRecProducers/plugins/DeclsForKernelsPh2.h

```diff
@@ -0,0 +1,28 @@
+#ifndef RecoLocalCalo_EcalRecProducers_plugins_DeclsForKernelsPh2_h
+#define RecoLocalCalo_EcalRecProducers_plugins_DeclsForKernelsPh2_h
+
+#include "CUDADataFormats/EcalRecHitSoA/interface/EcalUncalibratedRecHit.h"
+#include "DataFormats/EcalDigi/interface/EcalDataFrame_Ph2.h"
+
+namespace ecal {
+  namespace weights {
+
+    struct EventOutputDataGPU {
+      UncalibratedRecHit<::calo::common::DevStoragePolicy> recHits;
+
+      void allocate(uint32_t digi_size, cudaStream_t cudaStream) {
+        auto const size = digi_size;
+        recHits.amplitudesAll = cms::cuda::make_device_unique<reco::ComputationScalarType[]>(
+            size * EcalDataFrame_Ph2::MAXSAMPLES, cudaStream);
```

Comment:
As above, this seems strange. The number of uncalibrated RecHit amplitudes is not correlated with the number of samples.